File: src/dataset/cpp/spiral-matrix_exercism-toemm/claude/ClaudeSonnet4_spiral_matrix_v3.cpp
#include "spiral_matrix.h"

namespace spiral_matrix {

std::vector<std::vector<uint32_t>> spiral_matrix(uint32_t size) {
    if (size == 0) {
        return {};
    }
    
    std::vector<std::vector<uint32_t>> out;
    out.reserve(size);
    for (uint32_t i = 0; i < size; ++i) {
        out.emplace_back(size);
    }

    uint32_t k = 1;
    uint32_t top = 0;
    uint32_t bottom = size - 1;
    uint32_t left = 0;
    uint32_t right = size - 1;

    while (top <= bottom && left <= right) {
        for (uint32_t col = left; col <= right; ++col) {
            out[top][col] = k++;
        }
        if (++top > bottom) break;

        for (uint32_t row = top; row <= bottom; ++row) {
            out[row][right] = k++;
        }
        if (right-- == 0 || right < left) break;

        for (uint32_t col = right + 1; col-- > left;) {
            out[bottom][col] = k++;
        }
        if (bottom-- == 0 || bottom < top) break;

        for (uint32_t row = bottom + 1; row-- > top;) {
            out[row][left] = k++;
        }
        ++left;
    }

    return out;
}

}  // namespace spiral_matrix

File: src/dataset/python/spiral-matrix_exercism-ThomasZumsteg/claude/ClaudeSonnet4_spiral_matrix_v3.py
def spiral(size):
    if size <= 0:
        return []
    
    result = [[0] * size for _ in range(size)]
    
    top, bottom = 0, size - 1
    left, right = 0, size - 1
    num = 1
    
    while top <= bottom and left <= right:
        for col in range(left, right + 1):
            result[top][col] = num
            num += 1
        top += 1
        
        for row in range(top, bottom + 1):
            result[row][right] = num
            num += 1
        right -= 1
        
        if top <= bottom:
            for col in range(right, left - 1, -1):
                result[bottom][col] = num
                num += 1
            bottom -= 1
        
        if left <= right:
            for row in range(bottom, top - 1, -1):
                result[row][left] = num
                num += 1
            left += 1
    
    return result